oak_runtime/introspection_browser_client/components/Event/index.tsx

```diff
@@ -0,0 +1,110 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import React from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import introspectionEventsProto from '~/protoc_out/proto/introspection_events_pb';
+
+const useObjectAsDescriptionListStyles = makeStyles({
+  dl: {
+    display: 'inline',
+    '& dl': { '&:before': { content: '"{ "' }, '&:after': { content: '" }"' } },
+  },
+  dt: {
+    display: 'inline',
+    margin: 0,
+    '&:after': { content: '": "' },
+  },
+  dd: {
+    display: 'inline',
+    margin: 0,
+    '&:after': { content: '", "' },
+    '&:last-of-type': { '&:after': { content: '""' } },
+  },
+});
+
+function ObjectAsDescriptionList({ object }: { object: Object }) {
+  const classes = useObjectAsDescriptionListStyles();
+  return (
+    <dl className={classes.dl}>
+      {Object.entries(object).map(([key, value]) => (
+        <>
+          <dt className={classes.dt}>{key}</dt>
+          <dd className={classes.dd}>
+            {Array.isArray(value) ? (
+              JSON.stringify(value)
+            ) : typeof value === 'object' ? (
+              <ObjectAsDescriptionList object={value} />
+            ) : (
+              value
+            )}
```

Comment:
The array check needs to be before the object check, since 
```js
typeof [1, 2, 3]
// 'object'
```
😅

Reply:
Is it worth adding this as a comment in the file, or is it obvious enough for people who are not me?

Reply:
It's pretty well known JS behavior, but I added a comment to make the code friendlier to non JS contributors. :)